tools/changelog-helper.sh

```diff
@@ -0,0 +1,283 @@
+    local changelog=$(sed -rne '/^###.*'${target_release//./\.}'\b/,/^### '${prev_release//./\.}'\b/p' ChangeLog | tail -n +2 | head -n -1)
+    local changelog_orig="${changelog}"
+
+    # Is there an existing entry for this language already?
+    changelog=$(sed -re "s/^(  \* ${full_lang}, by .+ \(.*)\)/\1, #${id})/" <<<"${changelog}")
+    if [[ "${changelog}" == "${changelog_orig}" ]]; then
+        # No existing language entry. Check for an existing translation entry.
+        changelog=$(sed -re "s/^(- ${TRANSLATION_ENTRY_TEXT}.*)/\1\n  * ${full_lang}, by @${author} (#${id})/" <<<"${changelog}")
+        if [[ "${changelog}" == "${changelog_orig}" ]]; then
+            # No existing translation entry at all. Add a new one.
+            changelog="${changelog}$(
+                echo
+                echo
+                echo "- ${TRANSLATION_ENTRY_TEXT}"
+                echo "  * ${full_lang}, by @${author} (#${id})"
+			)"
+        else
+            # Existing translation entries, so sort them:
+            local changelog_before_translations=""
+            local changelog_translations=""
+            local changelog_after_translations=""
+            local changelog_translations_pos=before # before|block|after
+            while IFS= read -r line; do
+                if [[ "${changelog_translations_pos}" == "before" ]]; then
+                    if [[ "${line}" == "- ${TRANSLATION_ENTRY_TEXT}" ]]; then
+                        changelog_translations_pos=block
+                    fi
+                    changelog_before_translations="${changelog_before_translations}${line}"$'\n'
+                    continue
+                fi
+                if [[ "${changelog_translations_pos}" == "block" ]]; then
+                    if [[ "${line}" == "" ]]; then
+                        changelog_translations_pos=after
+                        # fallthrough
+                    else
+                        changelog_translations="${changelog_translations}${line}"$'\n'
+                        continue
+                    fi
+                fi
+                if [[ "${changelog_translations_pos}" == "after" ]]; then
+                    changelog_after_translations="${changelog_after_translations}${line}"$'\n'
+                fi
+            done <<< "${changelog}"
+            changelog="$(
+                # echo -n strips whitespace. we need that here.
+                echo -n "${changelog_before_translations}"
+                echo -n "$(grep -vP '^$' <<< "${changelog_translations}" | sort)"
+                echo -n "${changelog_after_translations}"
+            )"
+        fi
+    fi
+    # Rebuild the changelog and write back to file:
+    (echo "$changelog_header"; echo "$changelog"; echo; echo "$changelog_prev_releases") > ChangeLog
+}
+
+sanitize_title() {
+    sed \
+        -re 's/^\s+//' \
+        -re 's/\s{2,}/ /' \
+        -re 's/\s*\.?\s*$//' \
+        -re 's/\b((Add)|(Updat|Enhanc|Improv|Remove)e)\b/\2\3ed/i'
```

Comment:
Why is Remove not Remov here?

I didn’t fully analyse what this line does

Reply:
Thanks, fixed.